AccountingService/Controllers/UsersController.cs

```diff
@@ -15,130 +16,155 @@ public class UsersController : ControllerBase
         public IActionResult UpdateUser([FromBody] User request)
         {
-            salaryConversionContext.User.Update(request);
-            int changes = salaryConversionContext.SaveChanges();
-            if (changes == 0)
+            try
             {
-                return BadGateway();
+                salaryConversionContext.User.Update(request);
+                int changes = salaryConversionContext.SaveChanges();
+
+                return Ok();
             }
 
-            return Ok();
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error: " + ex.Message);
+                return new StatusCodeResult(500);
+            }
         }
 
         [HttpDelete("{id}")]
         public IActionResult DeleteUser(int id)
         {
-            var user = salaryConversionContext.User.Where(u => u.ID == id).SingleOrDefault();
-            if (user == null)
             {
-                return NotFound();
-            }
-            salaryConversionContext.User.Remove(user);
-            int changes = salaryConversionContext.SaveChanges();
-
-            if (changes == 0)
-            {
-                return BadGateway();
+                try
+                {
+                    var user = salaryConversionContext.User.Where(u => u.ID == id).SingleOrDefault();
+                    if (user == null)
+                    {
+                        return NotFound();
+                    }
+                    salaryConversionContext.User.Remove(user);
+                    int changes = salaryConversionContext.SaveChanges();
+
+
+                    return Ok();
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Error: " + ex.Message);
+                    return new StatusCodeResult(500);
+                }
             }
-            return Ok();
         }
 
-
-
         [HttpGet("{id}/salary")]
+
         public IActionResult GetUserSalary(int id, [FromQuery(Name = "currency")] string currency, [FromQuery(Name = "isNetSalary")] bool isNetSalary)
         {
-            double salary = salaryConversionContext.User
-                .Where(u => u.ID == id)
-                .Select(u => u.Salary)
-                .SingleOrDefault();
-
-            if (salary == 0)
+            try
             {
-                return NotFound();
-            }
+                double salary = salaryConversionContext.User
+                    .Where(u => u.ID == id)
+                    .Select(u => u.Salary)
+                    .SingleOrDefault();
 
-            double exchangeRate = exchangeRateService.GetCurrencyExchangeRate(currency);
-            salary = salary * exchangeRate;
 
-            if (isNetSalary)
-            {
-                double netSalary = netSalaryService.Calculate(salary);
+
+                double exchangeRate = exchangeRateService.GetCurrencyExchangeRate(currency);
+                salary = salary * exchangeRate;
```

Comment:
Everything after this line can not throw an exception. Move it out of the `try/catch` block.